packages/flutter/test/widgets/inherited_theme_test.dart

```diff
@@ -0,0 +1,137 @@
+                              ? InheritedTheme.captureAll(context, const IconTextBox('Hello'))
+                              : const IconTextBox('Hello')
+                          ),
+                        );
+                      },
+                      child: const IconTextBox('Tap'),
+                    );
+                  },
+                ),
+              ),
+            ),
+          );
+        },
+      );
+    }
+
+    TextStyle getIconStyle() {
+      return tester.widget<RichText>(
+        find.descendant(
+          of: find.byType(Icon),
+          matching: find.byType(RichText),
+        ),
+      ).text.style;
+    }
+
+    TextStyle getTextStyle(String text) {
+      return tester.widget<RichText>(
+        find.descendant(
+          of: find.text(text),
+          matching: find.byType(RichText),
+        ),
+      ).text.style;
+    }
+
+    useCaptureAll = false;
+    await tester.pumpWidget(buildFrame());
+    expect(find.text('Tap'), findsOneWidget);
+    expect(find.text('Hello'), findsNothing);
+    expect(getTextStyle('Tap').color, textColor);
+    expect(getTextStyle('Tap').fontSize, fontSize);
+    expect(getIconStyle().color, iconColor);
+    expect(getIconStyle().fontSize, iconSize);
+
+    // Tap to show the TestRoute
+    await tester.tap(find.text('Tap'));
+    await tester.pumpAndSettle(); // route transition
+    expect(find.text('Tap'), findsNothing);
+    expect(find.text('Hello'), findsOneWidget);
+    // The new route's text and icons will NOT inherit the DefaultTextStyle or
+    // IconTheme values.
+    expect(getTextStyle('Hello').color, isNot(textColor));
+    expect(getTextStyle('Hello').fontSize, isNot(fontSize));
+    expect(getIconStyle().color, isNot(iconColor));
+    expect(getIconStyle().fontSize, isNot(iconSize));
+
+    // Return to the home route
+    useCaptureAll = true;
+    Navigator.of(navigatorContext).pop();
+    await tester.pumpAndSettle(); // route transition
+
+    // Verify that all is the same as it was when the test started
+    expect(find.text('Tap'), findsOneWidget);
+    expect(find.text('Hello'), findsNothing);
+    expect(getTextStyle('Tap').color, textColor);
+    expect(getTextStyle('Tap').fontSize, fontSize);
+    expect(getIconStyle().color, iconColor);
+    expect(getIconStyle().fontSize, iconSize);
+
+    // Tap to show the TestRoute. The test route's IconTextBox will have been
+    // wrapped with InheritedTheme.captureAll().
+    await tester.tap(find.text('Tap'));
+    await tester.pumpAndSettle(); // route transition
+    expect(find.text('Tap'), findsNothing);
+    expect(find.text('Hello'), findsOneWidget);
+    // The new route's text and icons will inherit the DefaultTextStyle or
+    // IconTheme values because captureAll.
+    expect(getTextStyle('Hello').color, textColor);
+    expect(getTextStyle('Hello').fontSize, fontSize);
+    expect(getIconStyle().color, iconColor);
+    expect(getIconStyle().fontSize, iconSize);
+
+  });
+}
```

Comment:
Maybe verify that if there're multiple instances of `DefaultTextStyle`, the closest ancestor `DefaultTextStyle` wins?

Reply:
Good idea, will do that.